tests/zfs-tests/tests/functional/fault/auto_online_001_pos.ksh

```diff
@@ -0,0 +1,151 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2016 by Intel Corporation. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Tesing auto-online FMA ZED logic.
+#
+# Method: Creates a pool, exports pool, offlines disk,
+# imports pool with missing disk, then onlines disk.
+# ZED polls for an event change for online disk to be automatically
+# added back to the pool.
+# Creates a raidz1 zpool using persistent disk path names
+# (ie not /dev/sdc). Tests import using pool guid and cache file.
+#
+# Only works for real devices (not loopback)
+
+verify_runnable "both"
+
+disk1=${DISKS%% *}
+if ! ( is_real_device $disk1 || is_mpath_device $disk1 ); then
+	log_unsupported "This test can only be run on real disks."
+fi
+
+function cleanup
+{
+	#online last disk before fail
+	on_off_disk $offline_disk "online" $HBA $CHAN $TARG $LUN
+	poolexists $pool && destroy_pool $pool
+}
+
+log_assert "Testing auto-online FMA ZED logic"
+
+log_onexit cleanup
+
+pool=auto-online_test
+fault_cachefile=$STF_SUITE/tests/functional/fault/zpool_fault.cache
+target=$pool
+
+# Check to make sure ZED is running
+$SYSTEMCTL is-active zfs-zed > /dev/null
+if (($? != 0)); then
+	log_note "Starting ZED"
+	log_must $SYSTEMCTL start zfs-zed
+fi
```

Comment:
I'd like to avoid adding a dependency on systemd for these tests.  Adding this dependency rules out running the tests in-tree and on systems which don't use systemd.

Instead I suggest starting the ZED explicitly in the `setup.sh` and terminating it in `cleanup.sh`.  If the ZED is already running I'd suggest skipping this test group since we can't be sure the _right_ version of the ZED is running since we didn't start it.  Checking for `/var/run/zed.pid` is a easy way to do this.

Starting the ZED ourselves has a few advantages:

* We'll know for certain it's the version we expect.
* We can control the location of the ZEDLET directory, zed.pid, and zed.state files.
* We can run it in the foreground with the verbose flag and control the location of the log file.
* As features are added to the ZED it'll be easy to configure them for testing.